CI test runners sometimes do not have an available disk for OSDs

<!-- **Are you in the right place?**
1. For issues or feature requests, please create an issue in this repository.
2. For general technical and non-technical questions, we are happy to help you on our [Rook.io Slack](https://slack.rook.io/).
3. Did you already search the existing open issues for anything similar? -->

**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
The CI is failing frequently starting a couple days ago.

The tests are failing consistently during the "setup cluster resources" action, when attempting to detect and prepare the disks/partitions for the OSDs.
```
+ sudo udevadm control --log-priority=debug
+ sudo swapoff --all --verbose
swapoff /swapfile
+ sudo umount /mnt
umount: /mnt: not mounted.
Error: Process completed with exit code 32.
```

When connecting to a test runner to analyze the failure, there is only a single 150GiB disk:
```
$ lsblk
NAME    MAJ:MIN RM   SIZE RO TYPE MOUNTPOINTS
sda       8:0    0   150G  0 disk 
├─sda1    8:1    0 149.9G  0 part /
├─sda14   8:14   0     4M  0 part 
└─sda15   8:15   0   106M  0 part /boot/efi
```

When connecting to a runner that is passing the test, there are two 75GiB disks, then we re-partition one of them for the tests:
```
$ lsblk
NAME    MAJ:MIN RM  SIZE RO TYPE MOUNTPOINTS
sda       8:0    0   75G  0 disk 
├─sda1    8:1    0 74.9G  0 part /
├─sda14   8:14   0    4M  0 part 
└─sda15   8:15   0  106M  0 part /boot/efi
sdb       8:16   0   75G  0 disk 
├─sdb1    8:17   0    6G  0 part 
└─sdb2    8:18   0    6G  0 part 
```

We need to use only the runners that have the available second disk. 

**Expected behavior:**
Green CI

**How to reproduce it (minimal and precise):**
<!-- Please let us know any circumstances for reproduction of your bug. -->
See any recent integration tests such as the [Canary tests](https://github.com/rook/rook/actions/workflows/canary-integration-suite.yml).

--- a/tests/scripts/github-action-helper.sh
+++ b/tests/scripts/github-action-helper.sh
@@ -38,6 +38,21 @@ fi
 # FUNCTIONS #
 #############
 
+function create_extra_disk() {
+  sudo apt install -y targetcli-fb open-iscsi
+  truncate -s 75G ~/iscsi-disk.img
+  sudo targetcli /backstores/fileio create disk1 ~/iscsi-disk.img 75G
+  local target_iqn=iqn.2026-02.target.local:disk1
+  sudo targetcli /iscsi create ${target_iqn}
+  sudo targetcli /iscsi/${target_iqn}/tpg1/luns create /backstores/fileio/disk1
+  local init_iqn=iqn.2026-02.initiator.local
+  echo "InitiatorName=${init_iqn}" | sudo tee /etc/iscsi/initiatorname.iscsi >/dev/null
+  sudo targetcli /iscsi/${target_iqn}/tpg1/acls create ${init_iqn}
+  sudo targetcli /iscsi/${target_iqn}/tpg1/acls/${init_iqn} create tpg_lun_or_backstore=lun0 mapped_lun=0
+  sudo iscsiadm -m discovery -t sendtargets -p 127.0.0.1
+  sudo iscsiadm -m node --login
+}
+
 function find_extra_block_dev() {
   # shellcheck disable=SC2005 # redirect doesn't work with sudo, so use echo
   echo "$(sudo lsblk)" >/dev/stderr # print lsblk output to stderr for debugging in case of future errors
@@ -49,6 +64,10 @@ function find_extra_block_dev() {
   echo "  == find_extra_block_dev(): boot_dev='$boot_dev'" >/dev/stderr # debug in case of future errors
   # --nodeps ignores partitions
   extra_dev="$(sudo lsblk --noheading --list --nodeps --output KNAME | egrep -v "($boot_dev|loop|nbd)" | head -1)"
+  if [ -z "$extra_dev" ]; then
+    create_extra_disk >/dev/stderr
+    extra_dev="$(sudo lsblk --noheading --list --nodeps --output KNAME | egrep -v "($boot_dev|loop|nbd)" | head -1)"
+  fi
   echo "  == find_extra_block_dev(): extra_dev='$extra_dev'" >/dev/stderr # debug in case of future errors
   echo "$extra_dev"                                                       # output of function
 }
@@ -116,6 +135,14 @@ function use_local_disk_for_integration_test() {
   sudo apt purge snapd -y
   sudo udevadm control --log-priority=debug
   sudo swapoff --all --verbose
+
+  # Create an extra disk if doesn't exist.
+  : "$(block_dev)"
+  sudo lsblk
+
+  unset pipefail
+  mountpoint -q /mnt || return 0
+  set pipefail
   sudo umount /mnt
   sudo sed -i.bak '/\/mnt/d' /etc/fstab
   # search for the device since it keeps changing between sda and sdb

